src/software/power/control_executor.cpp

```diff
@@ -0,0 +1,94 @@
+#include "control_executor.h"
+
+#include <utility>
+
+ControlExecutor::ControlExecutor(std::shared_ptr<Charger> charger,
+                                 std::shared_ptr<Chicker> chicker,
+                                 std::shared_ptr<Geneva> geneva)
+    : charger(std::move(charger)), chicker(std::move(chicker)), geneva(std::move(geneva))
+{
+}
+
+void ControlExecutor::execute(const TbotsProto_PowerControl& control)
+{
+    switch (control.chicker.which_chicker_command)
+    {
+        case TbotsProto_PowerControl_ChickerControl_kick_speed_m_per_s_tag:
+            detachInterrupt(digitalPinToInterrupt(BREAK_BEAM_PIN));
```

Comment:
this line is duplicated in all cases, can you call it once at the top?

Reply:
I dont think we want it triggered in the default case when a chicker command isn't specified